packages/core/src/controllers/relayer.ts

```diff
@@ -395,6 +397,7 @@ export class Relayer extends IRelayer {
   private async createProvider() {
     if (this.provider.connection) {
       this.unregisterProviderListeners();
+      this.provider = undefined as any;
```

Comment:
This breaks the API of `provider`. 

This might cause bugs we'd need to track down later since everything dealing with provider expects `IJsonRpcProvider` and not something potentially `undefined`.